src/webpimage.cpp

```diff
@@ -827,8 +827,9 @@ namespace Exiv2 {
         }
     }
 
-    long WebPImage::getHeaderOffset(byte *data, long data_size,
-                                    byte *header, long header_size) {
+    long WebPImage::getHeaderOffset(byte* data, long data_size, byte* header, long header_size)
+    {
+        enforce(data_size >= header_size, kerFailedToReadImageData);
```

Comment:
How about this instead?

```suggestion
        if (data_size < header_size) { return -1; }
```

Solves the issue on master and does not require throwing another exception.

Reply:
Dan:  You are one really smart cookie.  You have a built-in sense for elegance.  And great at thinking about how something might fail to work.

Luis:  You're a really smart cookie as well of course.